src/plugins/mission_raw/mission_raw_impl.cpp

```diff
@@ -22,16 +22,36 @@ void MissionRawImpl::init()
         MAVLINK_MSG_ID_MISSION_ACK,
         std::bind(&MissionRawImpl::process_mission_ack, this, _1),
         this);
+
+    _parent->register_mavlink_message_handler(
+        MAVLINK_MSG_ID_MISSION_CURRENT,
+        std::bind(&MissionRawImpl::process_mission_current, this, _1),
+        this);
+
+    _parent->register_mavlink_message_handler(
+        MAVLINK_MSG_ID_MISSION_ITEM_REACHED,
+        std::bind(&MissionRawImpl::process_mission_item_reached, this, _1),
+        this);
 }
 
+void MissionRawImpl::enable() {}
+
+void MissionRawImpl::disable() {}
+
 void MissionRawImpl::deinit()
 {
+    _parent->unregister_timeout_handler(_timeout_cookie);
     _parent->unregister_all_mavlink_message_handlers(this);
 }
 
-void MissionRawImpl::enable() {}
-
-void MissionRawImpl::disable() {}
+void MissionRawImpl::reset_mission_progress()
+{
+    std::lock_guard<std::recursive_mutex> lock(_mission_data.mutex);
```

Comment:
Let's not make the mistake of using the `std::recursive_mutex` again. I don't think this should be required.

Reply:
Changed. Note this is taken from mission_impl.cpp. Should we update in MissionImpl:reset_mission_progress() as well then ?

Reply:
We should but we need to be careful and test it.